pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/PurgeMinionClusterIntegrationTest.java

```diff
@@ -0,0 +1,380 @@
+import org.apache.pinot.controller.helix.core.minion.PinotTaskManager;
+import org.apache.pinot.core.common.MinionConstants;
+import org.apache.pinot.segment.local.segment.creator.impl.SegmentIndexCreationDriverImpl;
+import org.apache.pinot.segment.spi.creator.SegmentGeneratorConfig;
+import org.apache.pinot.segment.spi.creator.SegmentIndexCreationDriver;
+import org.apache.pinot.spi.config.table.SegmentPartitionConfig;
+import org.apache.pinot.spi.config.table.SegmentZKPropsConfig;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableTaskConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.utils.builder.TableConfigBuilder;
+import org.apache.pinot.spi.utils.builder.TableNameBuilder;
+import org.apache.pinot.util.TestUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+
+/**
+ * Integration test for minion task of type "PurgeTask"
+ */
+public class PurgeMinionClusterIntegrationTest extends BaseClusterIntegrationTest {
+  private static final String PURGE_FIRST_RUN_TABLE = "myTable1";
+  private static final String PURGE_DELTA_PASSED_TABLE = "myTable2";
+  private static final String PURGE_DELTA_NOT_PASSED_TABLE = "myTable3";
+
+  protected PinotHelixTaskResourceManager _helixTaskResourceManager;
+  protected PinotTaskManager _taskManager;
+  protected PinotHelixResourceManager _pinotHelixResourceManager;
+
+  protected final File _segmentDir1 = new File(_tempDir, "segmentDir1");
+  protected final File _segmentDir2 = new File(_tempDir, "segmentDir2");
+  protected final File _segmentDir3 = new File(_tempDir, "segmentDir3");
+
+  protected final File _tarDir1 = new File(_tempDir, "tarDir1");
+  protected final File _tarDir2 = new File(_tempDir, "tarDir2");
+  protected final File _tarDir3 = new File(_tempDir, "tarDir3");
+
+  @BeforeClass
+  public void setUp()
+      throws Exception {
+    TestUtils.ensureDirectoriesExistAndEmpty(_tempDir, _segmentDir1, _tarDir1,
+        _segmentDir2, _tarDir2, _segmentDir3, _tarDir3);
+
+    // Start the Pinot cluster
+    startZk();
+    startController();
+    startBrokers(1);
+    startServers(1);
+
+    // Create and upload the schema and table config
+    Schema schema = createSchema();
+    addSchema(schema);
+    TableConfig purgeTableConfig =
+        createOfflineTableConfig(PURGE_FIRST_RUN_TABLE, getPurgeTaskConfig());
+    TableConfig purgeDeltaPassedTableConfig =
+        createOfflineTableConfig(PURGE_DELTA_PASSED_TABLE, getPurgeTaskConfig());
+    TableConfig purgeDeltaNotPassedTableConfig =
+        createOfflineTableConfig(PURGE_DELTA_NOT_PASSED_TABLE, getPurgeTaskConfig());
+    addTableConfig(purgeTableConfig);
+    addTableConfig(purgeDeltaPassedTableConfig);
+    addTableConfig(purgeDeltaNotPassedTableConfig);
+
```

Comment:
(minor) Remove the consecutive empty lines. Same for other unnecessary whitespaces in this class